keras_retinanet/plugins/coco/coco.dataset

```diff
@@ -0,0 +1,23 @@
+;Copyright 2017-2018 Ashley Williamson (https://inp.io)
+;
+;Licensed under the Apache License, Version 2.0 (the "License");
+;you may not use this file except in compliance with the License.
+;You may obtain a copy of the License at
+;
+;    http://www.apache.org/licenses/LICENSE-2.0
+;
+;Unless required by applicable law or agreed to in writing, software
+;distributed under the License is distributed on an "AS IS" BASIS,
+;WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;See the License for the specific language governing permissions and
+;limitations under the License.
+
+[Core]
+Name = Coco Dataset
```

Comment:
Would it be an idea to use name here to mean the name of the subparser? So in this case that would be `pascal`. Instead of an attribute called `dataset_type` on the plugin object we could then just use the plugin name to look them up.

Reply:
The spirit of the Info file denotes that as the name. I think it should remain as human readable, as it's intended use is for relaying back to users. The justification behind 'dataset_type' is so that we have a common ident for a given dataset. Eg, VOC is PASCAL VOC, yet we only use `pascal` as the ident. It allows Datasets to be correctly named ( with even more info ), then also self-identify their short-hand.

Reply:
On the other hand, finding out the command line argument for a plugin now requires to look at the source of the plugin. To me it makes sense if that information comes from the plugin description file.

Reply:
Well, ideally plugins would have their own documentation. However, because the subparser decorator adds entries to the subparser, if a user types an incorrect flag it should correct them to say it's not of type <csv|coco|voc>. I believe this was the case during my testing. Remember, these plugins should be able to exist separately from RetinaNet, perhaps residing as a pip package, etc. I think it's very important to keep high-level human readability in mind, like with every other package/plugin system.

Reply:
I don't want human readability to come at the cost of making it easy for humans to understand which plugin belongs to which parser. Considering that we already have a human readable description field, I think the name is a perfect candidate for being the link between the plugin file and the subparser  name.

Reply:
However, the name and description reside in the plugin domain not the RetinaNet domain. For instance, I may have multiple plugins which define 'coco' for me. Maybe taken from different years etc. With the plugin system I can programmatically enable/disable plugins and chose which are active which may be of use in test cases or scripts users may want to write. I don't believe description to be a suitable replacement for name, both are required together.

Reply:
I don't find that very compelling. If the user wants to enable/disable plugins they could do that by modifying the used plugin path, or like systemd does by creating/removing symlinks (even windows has those nowadays).

The user should not have to modify code to enable or disable plugins. To me that sort of defeats the purpose of having a plugin system.

The main thing a dataset plugin does is hook in on the creation of a dataset generator. At that place where that hooking occurs, it is identified by a string. So that string is already an identifier for the plugin. I don't want to have two different identifiers.

Slight sidestep: does yapsy allow access to the name of the plugin description file?